scripts/build_ou_vb.py

```diff
@@ -0,0 +1,245 @@
+        suffix = f"{acoustic_embed_suffix}"
+        subbanks.append({"color": color, "suffix": suffix})
+
+    if subbanks:
+        with open(f"{main_stuff}/character.yaml", "r") as config:
+            character_config = yaml.safe_load(config)
+        character_config["subbanks"] = subbanks
+        with open(f"{main_stuff}/character.yaml", "w") as config:
+            yaml.dump(character_config, config)
+
+    #image, portrait, and portrait opacity can be manually edited
+    with open(f"{main_stuff}/character.yaml", "a") as file:
+        file.write("\n")
+        file.write("text_file_encoding: utf-8\n")
+        file.write("\n")
+        file.write("image:\n")
+        file.write("portrait:\n")
+        file.write("portrait_opacity: 0.45\n")
+
+    with open(f"{main_stuff}/dsconfig.yaml", "w") as file:
+        file.write("phonemes: dsmain/phonemes.txt\n")
+        file.write("acoustic: dsmain/acoustic.onnx\n")
+        file.write("vocoder: nsf_hifigan\n")
+        file.write("singer_type: diffsinger\n")
+
+    with open(acoustic_config, "r") as config:
+        acoustic_config_data = yaml.safe_load(config)
+
+    use_energy_embed = acoustic_config_data.get("use_energy_embed")
+    use_breathiness_embed = acoustic_config_data.get("use_breathiness_embed")
+    use_shallow_diffusion = acoustic_config_data.get("use_shallow_diffusion")
+    max_depth = acoustic_config_data.get("K_step")
+    speakers = acoustic_config_data.get("speakers")
+    augmentation_arg = acoustic_config_data.get("augmentation_args")
+    pitch_aug = acoustic_config_data.get("use_key_shift_embed")
+    time_aug = acoustic_config_data.get("use_speed_embed")
+
+    with open(f"{main_stuff}/dsconfig.yaml", "r") as config:
+        dsconfig_data = yaml.safe_load(config)
+
+    dsconfig_data["use_energy_embed"] = use_energy_embed
+    dsconfig_data["use_breathiness_embed"] = use_breathiness_embed
+    dsconfig_data["use_shallow_diffusion"] = use_shallow_diffusion
+    dsconfig_data["max_depth"] = max_depth
+    dsconfig_data["augmentation_args"] = augmentation_arg
+    dsconfig_data["use_key_shift_embed"] = pitch_aug
+    dsconfig_data["use_speed_embed"] = time_aug
+
+    if subbanks:
+        dsconfig_data["speakers"] = acoustic_embeds
+
+    with open(f"{main_stuff}/dsconfig.yaml", "w") as config:
+        yaml.dump(dsconfig_data, config)
+
+    variance_emb_files = os.listdir(variance_onnx_folder)
+    variance_embeds = []
+
+    for file in variance_emb_files:
+        if file.endswith(".emb"):
+            variance_emb = os.path.splitext(file)[0]
+            variance_embeds.append("../dsmain/embeds/variance/" + variance_emb)
+    phoneme_dict_path = f"{acoustic_onnx_folder}/dictionary.txt"
+    dsdict = "dsdict.yaml"
+
+    def parse_phonemes(phonemes_str):
+        return phonemes_str.split()
+
+    entries = []
+    vowel_types = {"a", "i", "u", "e", "o", "N", "M", "NG", "cl", "vf", "AP", "SP"} #can be edited manually(?)

```

Comment:
To avoid conflicts with future updates, VCS-managed parts of the repo should not be edited by the user. Consider moving this to CLI argument or just let the user build the YAML dict elsewhere and simply pass its path here.